File: vinzi.jsonMgt/src/main/java/vinzi/jsonMgt/hib/actionEntry.java
package vinzi.jsonMgt.hib;

public class actionEntry {
    public Long id;
    public Long getId() { return id;}
    protected void setId(Long id)  {this.id = id; };


    public java.util.Date datetime;
    public java.util.Date getDatetime() { return datetime;}
    public void setDatetime(java.util.Date datetime)  {this.datetime = datetime; };


    public String track;
    public String getTrack() { return track;}
    public void setTrack(String track)  {this.track = track; };


    public String action;
    public String getAction() { return action;}
    public void setAction(String action)  {this.action = action; };


    public String user;
    public String getUser() { return user;}
    public void setUser(String user)  {this.user = user; };


    public actionEntry() {};

    public actionEntry(Long id, java.util.Date datetime, String track, String action, String user) {
	this.id = id;
	this.datetime = datetime;
	this.track = track;
	this.action = action;
	this.user = user;
	return;};

   public String toString() {
        return " id=" + this.id + " datetime=" + this.datetime + " track=" + this.track + " action=" + this.action + " user=" + this.user;};

};

File: vinzi.jsonMgt/src/main/clojure/vinzi/jsonMgt/hib/clj/actionEntry.clj
(ns vinzi.jsonMgt.hib.clj.actionEntry
    (:use vinzi.hib-connect.globals)
    (:import vinzi.jsonMgt.hib.actionEntry))

(defrecord actionEntry_clj [id datetime track action d_user])

(defn create-actionEntry [datetime track action d_user]
  ;; TO DO:  insert type-checking (java-objects are strictly typed)
  (actionEntry_clj. nil datetime track action d_user))

(defn clone-actionEntry [cRec]
  ;; TO DO:  insert type-checking cRec should be same type/class as target
  (create-actionEntry (:datetime cRec) (:track cRec) (:action cRec) (:d_user cRec)))

(defn actionEntry-to-java [cRec]
	(actionEntry. (:id cRec) (:datetime cRec) (:track cRec) (:action cRec) (:d_user cRec) ))


(add-to-java (actionEntry_clj. 0 1 2 3 4) actionEntry-to-java)

(defn actionEntry-to-clj [jRec]
	(actionEntry_clj. (.getId jRec) (.getDatetime jRec) (.getTrack jRec) (.getAction jRec) (.getD_user jRec) ))


(add-to-clj (actionEntry.) actionEntry-to-clj)

